nextcord/client.py

```diff
@@ -1643,3 +1646,32 @@ def persistent_views(self) -> Sequence[View]:
         .. versionadded:: 2.0
         """
         return self._connection.persistent_views
+
+    async def on_connect(self):
+        await self.get_application_commands()
+
+    async def on_interaction(self, interaction: Interaction):
+        # TODO: This doesn't seem optimal, I need reassurance?
+        if interaction.type is InteractionType.ping:
+            self.dispatch("ping", interaction)
+        elif interaction.type is InteractionType.application_command:
+            self.dispatch("application_command", interaction)
+        elif interaction.type is InteractionType.component:
+            self.dispatch("component", interaction)
```

Comment:
Possibly still could be done better, but some repetition could be avoided with a dict 

```suggestion
        events = {
        	InteractionType.ping: "ping",
        	InteractionType.application_command: "application_command",
        	InteractionType.component: "component"
        }
        self.dispatch(events[interaction.type], interaction)
```